dd-java-agent/instrumentation/jax-rs-annotations/src/main/java/datadog/trace/instrumentation/jaxrs/JaxRsAnnotationsInstrumentation.java

```diff
@@ -135,21 +122,54 @@ public static Scope nameSpan(@Advice.Origin final Method method) {
 
       final String operationName = className + "." + methodName;
 
-      return GlobalTracer.get()
-          .buildSpan(operationName)
-          .withTag(Tags.COMPONENT.getKey(), "jax-rs-controller")
-          .startActive(true);
+      final Stack<Scope> scopeStack = new Stack<>();
+
+      Scope parentScope = GlobalTracer.get().scopeManager().active();
+      if (parentScope == null) {
+        parentScope =
+            GlobalTracer.get()
+                .buildSpan(operationName)
+                .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
+                .withTag(Tags.COMPONENT.getKey(), "jax-rs")
+                .startActive(true);
+        scopeStack.push(parentScope);
+      }
+
+      if (!resourceName.isEmpty()) {
+        Span span = parentScope.span();
+        span.setOperationName(resourceName);
+        Tags.SPAN_KIND.set(span, Tags.SPAN_KIND_SERVER);
```

Comment:
I think we should always set those tags (kind, component, `http.url` and http.method`) even if resourceName is empty (and also avoid the duplication between here and line 132).

Reply:
I went back and forth on this and thought the all-or-nothing was cleaner in some regards but think I agree overall.  Will update.